### DISCUSSION TITLE
Openid with mongo db collection

### DISCUSSION THREAD

**Discussion by vitalybrandes** [OrchardCMS/OrchardCore#11371] [Q&A] [answered]
Hi,
is there any example of how to setup openid module with with mongodb instead default orchard core db?

found this one 
https://kevinchalet.com/2020/10/03/using-the-orchardcore-openid-management-feature-with-an-existing-openiddict-deployment/
but it is for UseEntityFrameworkCore

**Comment by hishamco** [comment]
You can rewrite the module by replacing the underlying store, @kevinchalet is there a service that you can implement to change the store from the DI?

**Comment by hishamco** [comment]
@vitalybrandes I think you have to look into `UseEntityFrameworkCore()` in the call

```
services.AddOpenIddict()
    .AddCore()
    .UseEntityFrameworkCore()
    .UseDbContext<MyContext>();
```

**Reply by vitalybrandes** [reply]
> @vitalybrandes I think you have to look into `UseEntityFrameworkCore()` in the call
> 
> ```
> services.AddOpenIddict()
>     .AddCore()
>     .UseEntityFrameworkCore()
>     .UseDbContext<MyContext>();
> ```

I found this one in @kevinchalet blog
https://kevinchalet.com/2020/07/08/openiddict-3-0-beta2-is-out/
wondering if there is example with orchard :-)

**Reply by hishamco** [reply]
OC is using YesSQL not MangoDB

**Reply by vitalybrandes** [reply]
> OC is using YesSQL not MangoDB

Yeah, i know,
question if i can combine ,using mongo for only openid module.

**Comment by kevinchalet** [comment, marked as answer]
OpenIddict natively supports MongoDB: https://www.nuget.org/packages/OpenIddict.MongoDb/

The MongoDB integration is registered exactly the same way as the EF Core one shown in the blog post you mentioned.

**Reply by vitalybrandes** [reply]
@kevinchalet  Will this also support orchard core openid GUI updates?

**Reply by vitalybrandes** [reply]
@kevinchalet 
do you have any example how to use MongoDB  instead Yesql with orchard?
I cant find how to set it up.
```

            services.AddOrchardCms().AddSetupFeatures("OrchardCore.AutoSetup")
                    .ConfigureServices(services =>
                    {
                        services.AddOpenIddict()

                        // Register the OpenIddict core services.
                        .AddCore(options =>
                        {
                            // Configure OpenIddict to use the MongoDB stores and models.
                            options.UseMongoDb()
                                    .UseDatabase(new MongoClient("mongodb://localhost:27018").GetDatabase("openiddict"))
                                    .SetApplicationsCollectionName("openiddict.applications")
                                    .SetAuthorizationsCollectionName("openiddict.authorizations")
                                    .SetScopesCollectionName("openiddict.scopes")
                                    .SetTokensCollectionName("openiddict.tokens");
                        });
                    });
```

**Reply by vitalybrandes** [reply]
> @kevinchalet do you have any example how to use MongoDB instead Yesql with orchard? I cant find how to set it up.
> 
> ```
> 
>             services.AddOrchardCms().AddSetupFeatures("OrchardCore.AutoSetup")
>                     .ConfigureServices(services =>
>                     {
>                         services.AddOpenIddict()
> 
>                         // Register the OpenIddict core services.
>                         .AddCore(options =>
>                         {
>                             // Configure OpenIddict to use the MongoDB stores and models.
>                             options.UseMongoDb()
>                                     .UseDatabase(new MongoClient("mongodb://localhost:27018").GetDatabase("openiddict"))
>                                     .SetApplicationsCollectionName("openiddict.applications")
>                                     .SetAuthorizationsCollectionName("openiddict.authorizations")
>                                     .SetScopesCollectionName("openiddict.scopes")
>                                     .SetTokensCollectionName("openiddict.tokens");
>                         });
>                     });
> ```

GET IT!! THANKS
               services.AddOrchardCms()
                .ConfigureServices(async services =>
                {

                    services.AddOpenIddict()

                    // Register the OpenIddict core services.
                    .AddCore(options =>
                    {
                        // Configure OpenIddict to use the MongoDB stores and models.
                        options.UseMongoDb()
                                .UseDatabase(new MongoClient("mongodb://localhost:27018").GetDatabase("openiddict"));
                    });
                }, order: 10000);
